### ISSUE TITLE
CombiTable1Ds gives different results when multiple instances are used in MSL 3.2.1 beta1

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1150]
**Modified by dietmarw on 31 May 2013 06:53 UTC**
In the attached model, datRea.y![1] should be as shown in withMSL3.2.png:

```imagebox
[[Image(withMSL3.2.png)]]
```

If this model is run with MSL 3.2.1 beta 1, revision 6557, then the output is as shown in withMSL3.2.1beta1.png:

```imagebox
[[Image(withMSL3.2.1beta1.png)]]
```

If the instance datRea1 is deleted, then the output datRea.y![1] is correct. Also, if Modelica/Blocks/{Sources.mo, Tables.mo} are replaced with the version of MSL 3.2, then the results are correct.

The simulations were run with Dymola 2014 on Windows.

----

**Reported by mwetter on 30 May 2013 23:21 UTC**
In the attached model, datRea.y![1] should be as shown in withMSL3.2.png. If this model is run with MSL 3.2.1 beta 1, revision 6557, then the output is as shown in withMSL3.2.1beta1.png.

If the instance datRea1 is deleted, then the output datRea.y![1] is correct. Also, if Modelica/Blocks/{Sources.mo, Tables.mo} are replaced with the version of MSL 3.2, then the results are correct.

The simulations were run with Dymola 2014 on Windows.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1150

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 31 May 2013 06:53 UTC**
Embedded the images.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 31 May 2013 07:24 UTC**
I cannot reproduce it in SimulationX. Results are always as expected (upper diagram) for old and new tables.

Unfortunately I have no chance to debug table implementation with Dymola.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 31 May 2013 08:05 UTC**
I tested it in OpenModelica (with the patch from #1143), and it works fine since it's stored on file. I used trunk and not beta1 though.
```imagebox
 [[Image(OpenModelica.svg,500px)]]
```

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 31 May 2013 09:29 UTC**
I can see that there is an issue in Dymola, I could not see any obvious issues in the generated C-code - and no obvious relevant issues in the C-code of the tables except for the same issue with shallow copy as in #1143.
[So I will first apply the patch from that ticket before further testing.]

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 31 May 2013 11:59 UTC**
After applying the patch from #1143 I got the expected result in Dymola.
It seems these two tickets are the same; and further testing will wait until ticket #1143 is fully corrected.

*: Well, manually applying since my patch program did not recognize the patch-files.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 31 May 2013 12:09 UTC**
Hans, thanks for confirmation that ticket #1143 really is an issue that needs to be fixed.

Regarding this (#1150) ticket there is no need to wait for #1143 to be fully corrected as tables are stored on file.

Thus I suggest to commit the patch of #1143 and close this (#1150) ticket as duplicate. Is this fine with you?

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 31 May 2013 12:15 UTC**
Yes, closing this as a duplicate of #1143 seems fine.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 31 May 2013 12:28 UTC**
Resolved by a7f550e3c8a744bed76afbf966cf0c01f00cbe00.
